lib/Conversion/TTKernelToEmitC/TTKernelToEmitC.cpp

```diff
@@ -187,6 +191,46 @@ class TTMetalToEmitCOpaqueRewriter : public OpConversionPattern<SourceOp> {
     return name;
   }
 
+  template <typename ReduceKindOp>
+  std::pair<StringRef, StringRef> getReduceTypeAndDim(ReduceKindOp op) const {
+    StringRef reduceType =
+        op.getReduceTypeAttr().getValue() == ttkernel::ReduceType::ReduceMAX
+            ? "PoolType::MAX"
+            : "PoolType::SUM";
+    StringRef reduceDim =
+        op.getReduceDimAttr().getValue() == ttkernel::ReduceDim::ReduceDimCOL
+            ? "ReduceDim::REDUCE_COL"
+        : op.getReduceDimAttr().getValue() == ttkernel::ReduceDim::ReduceDimROW
+            ? "ReduceDim::REDUCE_ROW"
+            : "ReduceDim::REDUCE_SCALAR";
+    return {reduceType, reduceDim};
+  }
```

Comment:
I think autogenerated `stringify` functions might help with this. Check `build/include/ttmlir/Dialect/TTKernel/IR/TTKernelOpsEnums.h.inc`.

Reply:
Yeah there is function, but these values are copied from tt-metal. I am not sure if we want to embed them into our types in tablegen. It would look like:

```
def TTKernel_ReduceDimRow : I32EnumAttrCase<"Row", 0, "ReduceDim::REDUCE_ROW">;
```

and

```
def TTKernel_ReduceSum : I32EnumAttrCase<"Sum", 0, "PoolType::SUM">;
```

Not sure.

Reply:
Yeah, not nice but whatever, I just thought you might automatically get these strings from `op`, avoiding `getReduceTypeAndDim`, but `PoolType::SUM` is ugly.